packages/grid/x-grid-data-generator/src/useDemoData.ts

```diff
@@ -82,7 +82,7 @@ export const useDemoData = (options: DemoDataOptions): DemoDataReturnType => {
   const [data, setData] = React.useState<GridData>({ columns: [], rows: [] });
   const [rowLength, setRowLength] = React.useState(options.rowLength);
   const [index, setIndex] = React.useState(0);
-  const [loading, setLoading] = React.useState(false);
+  const [loading, setLoading] = React.useState(true);
```

Comment:
I think that initially the rows are not loading. And the grid should be empty
And only when you get in the useEffect it starts loading.
Have you tried to remove the useEffect?

Reply:
As far as I understand the `use.Effect` and what is inside is to prevent the freeze when fetching a lot of data. For me both having it by default as `true` and `false` are logical it depends on how we want to look at it.